peepingtom/core/datablocks/dipoleblock.py

```diff
@@ -0,0 +1,79 @@
+import numpy as np
+
+from .base import GroupBlock
+from .pointblock import PointBlock
+from .orientationblock import OrientationBlock
+from peepingtom.utils.helpers.linalg_helper import align_vectors
+
+
+class DipoleBlock(GroupBlock):
+    """
+    A DipoleBlock represents a set of n dipoles in an m-dimensional space
+    defined by center points and end points
+    """
+
+    def __init__(self, centers: PointBlock, endpoints: PointBlock):
+        # Set dipole info
+        self.centers = PointBlock(centers)
+        self.endpoints = PointBlock(endpoints)
+
+        # Check endpoint shape matches that of centers centers
+        self._check_shapes()
+
+        # Set blocks for GroupBlock
+        children = [centers, endpoints]
```

Comment:
This should take `self.centers` and `self.endpoints`. You want to pass the datablocks, not the un-instanced data!

Reply:
gotcha!